### ISSUE TITLE
Add `physicalFilename` option into rule tester

### ISSUE THREAD

**Issue by JounQin** [eslint/eslint#14800]
@snitin315 @nzakas @mdjermanovic

I find `physicalFilename` is not available in `ValidTestCase`, so it makes testing cases for virtual filename unavailable, should that be supported?

Before `context.getPhysicalFilename`, I have a custom util `getPhysicalFilename` and a test case like:

```js
ruleTester.run('remark', remark, {
  valid: [
    {
      code: '<header>Header2</header>',
      parser,
      parserOptions,
      filename: path.resolve(__filename, '0-fake.mdx'), // virtual filename
    },
  ],
})
```

But when I migrate to `context.getPhysicalFilename`, it results `context.getPhysicalFilename() === filename`, it is not correct, so maybe a new `physicalFilename` option is required for this case:

```js
ruleTester.run('remark', remark, {
  valid: [
    {
      code: '<header>Header2</header>',
      parser,
      parserOptions,
      physicalFilename: __filename,
      filename: path.resolve(__filename, '0-fake.mdx'), // virtual filename
    },
  ],
})
```

_Originally posted by @JounQin in https://github.com/eslint/eslint/issues/14616#issuecomment-878712207_

**Comment by nzakas** [comment]
That seems reasonable to me. @eslint/eslint-tsc thoughts?

**Comment by mdjermanovic** [comment]
RuleTester uses [Linter API](https://eslint.org/docs/developer-guide/nodejs-api#linter). `physicalFilename` is not an option on the API. Values returned by `context.getFilename()` and `context.getPhysicalFilename()` are calculated from the passed `filename` and the output of processors applied during the process.

I think it would make more sense to support processors in RuleTester (https://github.com/eslint/rfcs/pull/31), than adding an option that would force Linter to change its internal logic and return some other value from `context.getPhysicalFilename()`.

**Comment by nzakas** [comment]
@mdjermanovic nice catch! That makes sense to me. I completely forgot about the RFC.

**Comment by nzakas** [comment]
**TSC Summary:** This issue requests to add a physicalFilename option to RuleTester to help with testing code blocks. There is already an RFC for enabling processors in RuleTester (https://github.com/eslint/rfcs/pull/31) that may be more appropriate.

** TSC Question:** How do we want to proceed here?

**Comment by btmills** [comment]
In the most recent TSC meeting, we agreed to go with the approach described in RFC31. I'll be picking up that RFC to update it as necessary and get it through approval.

**Comment by nzakas** [comment]
@btmills where are we on this?

**Comment by btmills** [comment]
RFC31 has been updated. Give it a look and please share any feedback!

**Comment by nzakas** [comment]
Closing due to age.
